src/shims/tls.rs

```diff
@@ -357,7 +357,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             // This is the first time we got asked to schedule a destructor. The
             // Windows schedule destructor function must be called exactly once,
             // this is why it is in this block.
-            if this.tcx.sess.target.os == "windows" {
+            if this.tcx.sess.target.os == "windows" && std::env::var_os("MIRI_NO_STD").is_none() {
```

Comment:
Also I think it'd be better to just change `schedule_windows_tls_dtors` to bail out in a controlled way when std is missing, instead of relying on the env var.